Data_Structures_Answers.md

```diff
@@ -1,18 +1,25 @@
 Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+O(n) - driven by the for-loop search for the oldest item
 
 2. What is the space complexity of your ring buffer's `append` function?
+O(1) - adds 1 item to the list each time
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+O(n) - driven by the check of whether each item is a None or not
 
 4. What is the space complexity of your ring buffer's `get` method?
-
+O(n) - creates and returns a new filtered array with the list comprehension
```

Comment:
`O(n)` in the worst case which is of course correct, since the newly-allocated list contains every element in the buffer. However, it is entirely possible to implement this method such that it exhibits O(1) space complexity.